app/src/main/java/com/halim/adam/cashmaster/AddNewSpending.java

```diff
@@ -4,17 +4,96 @@
 import android.content.Intent;
 import android.os.Bundle;
 import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.EditText;
+import android.widget.Spinner;
+
+import com.halim.adam.cashmaster.Objects.Budget;
+import com.halim.adam.cashmaster.Objects.BudgetRatio;
+import com.halim.adam.cashmaster.Objects.Category;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
 
 public class AddNewSpending extends Activity {
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd");
+    Spinner spinnerCategory;
+    Spinner spinnerBudget;
+    ArrayList<Integer> categoryIdList = new ArrayList<>();
+    ArrayList<Integer> budgetIdList = new ArrayList<>();
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_add_new_spending);
+        DatabaseHelper dbHelper = new DatabaseHelper(this);
+
+        // Spinners
+        spinnerCategory = findViewById(R.id.spinnerCategory);
+        spinnerBudget = findViewById(R.id.spinnerBudget);
+
+        // get data for spinner
+        ArrayList<Category> categoryList = dbHelper.GetCategoryList();
+        ArrayList<BudgetRatio> budgetRatioList = dbHelper.GetBudgetRatioList();
+
+        // insert data to string arraylist
+        ArrayList<String> categoryStrings = new ArrayList<>();
+        ArrayList<String> budgetStrings = new ArrayList<>();
+        for(int c = 0; c < categoryList.size(); c++){
+            categoryIdList.add(categoryList.get(c).getId());
+            categoryStrings.add(categoryList.get(c).getName());
+        }
+        for(int c = 0; c < budgetRatioList.size(); c++){
+            budgetIdList.add(budgetRatioList.get(c).getId());
+            budgetStrings.add(budgetRatioList.get(c).getName());
+        }
+
+        ArrayAdapter categoryAdapter = new ArrayAdapter(this, android.R.layout.simple_spinner_item, categoryStrings);
+        ArrayAdapter budgetAdapter = new ArrayAdapter(this, android.R.layout.simple_spinner_item, budgetStrings);
+        categoryAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        budgetAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+
+        spinnerCategory.setAdapter(categoryAdapter);
+        spinnerBudget.setAdapter(budgetAdapter);
     }
 
-    public void GetInput(){
+    public void GetInput(View view) throws ParseException {
+        EditText inputName = findViewById(R.id.inputName);
```

Comment:
hmm this is a bit uncommon I would say. 

Doing this, *finding views and reference it in a variable are usually done on the `onCreate` method, so that it will only be done once: During the activity loading.

If done heree, what happen is each time you call this function, you have to do the work to find the view and reference it to a variable once again